File: mycollab-mobile/src/main/scala/com/mycollab/mobile/module/project/events/ProjectMemberEvent.scala
package com.mycollab.mobile.module.project.events

import com.mycollab.events.ApplicationEvent

import scala.beans.BeanProperty

/**
  * @author MyCollab Ltd
  * @since 5.2.5
  */
object ProjectMemberEvent {

  @SerialVersionUID(1L)
  class InviteProjectMembers(@BeanProperty val inviteEmails: java.util.List[String], @BeanProperty val roleId: Integer,
                             @BeanProperty val inviteMessage: String) extends Serializable {}

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoInviteMembers(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/module/project/event/ProjectMemberEvent.kt
package com.mycollab.mobile.module.project.event

import com.mycollab.events.ApplicationEvent

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class ProjectMemberEvent {
    class InviteProjectMembers(val inviteEmails: List<String>, val roleId: Int,
                               val inviteMessage: String)

    class GotoList(source: Any, data: Any?) : ApplicationEvent(source, data)

    class GotoInviteMembers(source: Any, data: Any?) : ApplicationEvent(source, data)

    class GotoRead(source: Any, data: Any?) : ApplicationEvent(source, data)

    class GotoEdit(source: Any, data: Any?) : ApplicationEvent(source, data)
}